runtime/src/epoch_reward_calc_service.rs

```
@@ -0,0 +1,741 @@
+                measure!(bank.load_reward_calc_info(&thread_pool, &mut metrics));
+            calc_metrics.load_calc_info_time_us = load_calc_info_time.as_us();
+
+            let (signature, calc_signature_time) =
+                measure!(bank.compute_rewards_calc_signature(&vote_with_stake_delegations_map));
+            calc_metrics.calc_signature_time_us = calc_signature_time.as_us();
+
+            self.results
+                .write()
+                .unwrap()
+                .signatures
+                .insert(parent_slot, signature);
+
+            if !self
+                .results
+                .read()
+                .unwrap()
+                .rewards
+                .contains_key(&signature)
+            {
+                // hasn't seen the signature in the rewards yet, compute the rewards and save
+                // (signature, rewards) to the map.
+                let parent_epoch = epoch.saturating_sub(1); // TODO
+
+                let (result, reward_calc_time) = measure!(
+                    bank.do_stake_reward_calc(
+                        vote_with_stake_delegations_map,
+                        parent_epoch,
+                        &thread_pool,
+                        &mut metrics,
+                    ),
+                    "stake_reward_calc",
+                );
+                self.results
+                    .write()
+                    .unwrap()
+                    .rewards
+                    .insert(signature, Arc::new(result));
+                calc_metrics.reward_calc_time_us = reward_calc_time.as_us();
+            }
+
+            total_time.stop();
+
+            calc_metrics.total_time_us = total_time.as_us();
+            calc_metrics.report();
+        }
+    }
+
+    /// Handle incoming calculation requests loop
+    pub fn handle_request(&self) {
+        const MAX_REQUEST_RECV_TIME_MS: u64 = 500;
+        if let Ok(request) = self
```

this should be handled directly in the loop so we can break on error.  discarding the error like this will very likely cause the app to hang on exit and/or allow the service to silently die in the background.  neither of which are desirable operationally